File: src/UE4/FPakCompressedBlock.java
/**
 * 
 */
package UE4;

import UE4_Packages.FArchive;
import UE4_Packages.ReadException;

/**
 * @author FunGames
 *
 */
public class FPakCompressedBlock {
	public long compressedStart;
	public long compressedEnd;

	public FPakCompressedBlock(FArchive Ar) throws ReadException {
		compressedStart = Ar.readInt64();
		compressedEnd = Ar.readInt64();
	}
}

File: src/main/kotlin/me/fungames/jfortniteparse/ue4/pak/objects/FPakCompressedBlock.kt
package me.fungames.jfortniteparse.ue4.pak.objects

import me.fungames.jfortniteparse.ue4.reader.FArchive
import me.fungames.jfortniteparse.ue4.writer.FArchiveWriter

class FPakCompressedBlock {
    var compressedStart : Long
    var compressedEnd : Long

    constructor(Ar: FArchive) {
        compressedStart = Ar.readInt64()
        compressedEnd = Ar.readInt64()
    }

    fun serialize(Ar: FArchiveWriter) {
        Ar.writeInt64(compressedStart)
        Ar.writeInt64(compressedEnd)
    }

    constructor(compressedStart: Long, compressedEnd: Long) {
        this.compressedStart = compressedStart
        this.compressedEnd = compressedEnd
    }
}